OneCommunityGlobal/HighestGoodNetworkApp
# Aaryaneil - Unit Tests for userProfileByIdReducer

# Description
Unit test for `src/reducers/userProfileByIdReducer.test.js`




## Main changes explained:
Added test cases for the following:
 I. userProfileByIdReducer:
    1. should return the initial state
    2. should handle GET_USER_PROFILE
    3. should handle EDIT_USER_PROFILE
    4. should handle CLEAR_USER_PROFILE
 II. userTaskByIdReducer:
    1. should return the initial state
    2. should handle GET_USER_TASKS
 III. updateObject utility function:
    1. should update an object with new properties
    2. should not mutate the original object



## How to test:
1. check into current branch
6. do `npm install` and `npm test userProfileByIdReducer.test.js` to run this PR locally


## Screenshots or videos of changes:
<img width="1134" alt="Screenshot 2024-12-31 at 5 23 08 PM" src="https://github.com/user-attachments/assets/3455db21-87ae-4f82-b5a2-60e5a14c911f" />
